background.ts

```diff
@@ -0,0 +1,80 @@
+import { getMessageStreams } from "./src/components/utils/api.js"
+
+// remember the last timestamp for each stream
+const lastSeen: Record<string, number> = {};
+
+async function getStoredData(): Promise<{ token: string|null; locationId: string|null }> {
+  return new Promise((resolve) => {
+    chrome.storage.local.get(["token","locationId"], (res) => {
+      console.log("Background read token/locationId:", res.token, res.locationId);
+      resolve({ token: res.token ?? null, locationId: res.locationId ?? null });
+    });
+  });
+}
+
+// polling function
+async function checkForNewMessages() {
+  const { token, locationId } = await getStoredData();
+  if (!token || !locationId) return; 
+
+  // fetch streams for whatever location
+  const streams = await getMessageStreams(token, locationId);
+  streams.forEach((stream) => {
+    const timeStamp = new Date(stream.lastMessageAt).getTime();
+    const prev = lastSeen[stream.id] || 0;
+
+    if (timeStamp > prev) {
+      // new message
+      chrome.notifications.create(stream.id, {
+        type:           "basic",
+        iconUrl:        "icons/notif.png",
+        title:          `New message from ${stream.clientName}`,
+        message:        stream.lastMessage,
+        contextMessage: `${Math.floor((Date.now() - timeStamp) / 60000)}m ago`,
+        priority:       2,
+      });
+      lastSeen[stream.id] = timeStamp;
+    }
+  });
+}
+
+
+chrome.runtime.onInstalled.addListener(() => {
+  chrome.alarms.create("pollMessages", { periodInMinutes: 0.5 });
+});
+chrome.alarms.onAlarm.addListener(alarm => {
+  if (alarm.name === "pollMessages") checkForNewMessages();
+});
+
+// if the user clicks the notification, open popup and tell which stream to show
+chrome.notifications.onClicked.addListener((streamId) => {
+  //  construct a URL for popup page, including the stream ID
+  const url = chrome.runtime.getURL(`index.html#/dm?streamId=${streamId}`);
+  
+  // open it in a new popup-style window:
+  chrome.windows.create({
```

Comment:
1. Is there a way where this doesn't open a popup window, but rather the extension itself? 
2. If I already have a window open, another notification comes in, and I click it, I now have 2 windows open. Kind of a weird user experience. Is there a way around this? If we could implement the first bullet above, we might avoid this situation entirely. 

<img width="709" alt="image" src="https://github.com/user-attachments/assets/5d19fa7c-fdc4-47f7-8721-e0e1094357b6" />